soroban-sdk/src/env.rs

```diff
@@ -395,12 +397,14 @@ impl Env {
                     return;
                 }
                 // Allow if the last debug log entry contains the status of requested.
-                if let Some(HostEvent::Debug(dbg)) = clone.env_impl.get_events().0.last() {
-                    for arg in dbg.args.iter() {
-                        if let DebugArg::Val(v) = arg {
-                            if let Ok(st) = TryInto::<Status>::try_into(*v) {
-                                if st == status {
-                                    return;
+                if let Some(events) = clone.env_impl.get_events().ok().map(|e| e.0) {
+                    if let Some(HostEvent::Debug(dbg)) = events.last() {
```

Comment:
@jayz22 @graydon I had to update this SDK code when upgrading the host env, but I'm not confident that I completely understand the subtleties of what should happen in the error case of the `.get_events()` call. I'm calling .ok() here and just ignoring any issue with getting events. If it is fine, 🎉, if this isn't the appropriate behavior please lmk.